python/lsst/pipe/tasks/computeExposureSummaryStats.py

```diff
@@ -659,6 +663,63 @@ def update_effective_time_stats(self, summary, exposure):
         summary.effTimeSkyBgScale = float(b_eff)
         summary.effTimeZeroPointScale = float(c_eff)
 
+    def update_magnitude_limit_stats(self, summary, exposure, snr=5):
+        """Compute a summary statistic for the point-source magnitude limit at
+        a given signal-to-noise ratio using exposure-level metadata.
+
+        The magnitude limit is calculated at a given SNR from the PSF,
+        sky, zeropoint, and readnoise in accordance with SMTN-002 [1]_,
+        LSE-40 [2]_, and DMTN-296 [3]_.
+
+        References
+        ----------
+
+        .. [1] "Calculating LSST limiting magnitudes and SNR" (SMTN-002)
+        .. [2] "Photon Rates and SNR Calculations" (LSE-40)
+        .. [3] "Calculations of Image and Catalog Depth" (DMTN-296)
+
+
+        Parameters
+        ----------
+        summary  : `lsst.afw.image.ExposureSummaryStats`
+            Summary object to update in-place.
+        exposure : `lsst.afw.image.ExposureF`
+            Exposure to grab band and exposure time metadata
+        snr      : `float`, optional
+        """
+        if exposure.getDetector() is None:
+            summary.magLim = float("nan")
+            return
+
+        # Calculate the average readnoise [e-]
+        readNoiseList = list(ipIsr.getExposureReadNoises(exposure).values())
+        readNoise = np.nanmean(readNoiseList)
+        if np.isnan(readNoise):
+            readNoise = 0.0
+
+        # Calculate the average gain [e-/ADU]
+        gainList = list(ipIsr.getExposureGains(exposure).values())
+        gain = np.nanmean(gainList)
+        if np.isnan(gain):
+            gain = 1.0
```

Comment:
I agree that we should add a warning. My choice of the default value was intended to follow the prescription in isrTask [here](https://github.com/lsst/ip_isr/blob/c1d8f0290d0bbd0d32dcb30a8f9e67faa9475539/python/lsst/ip/isr/isrTask.py#L1882-L1886).

Reply:
Let me know if you still think we should bomb out with a NaN. I can see the argument for doing that, but I don't have a sense for what the DM standard is.

Reply:
Hmmm, yeah, I don't know. My _personal_ opinion is that having nothing is better than being wrong, and especially so if it's something that sufficiently subtly wrong that it can masquerade as being right, but I guess we should leave this to maybe @yalsayyad or someone to decide?

Reply:
Returning NaN sounds reasonable to me (that is done several other places in computeExposureSummaryStats). Also sounds like there is agreement from Yusra.

Reply:
Jumping in here:  Defaulting to 1.0 for the gain is usually the least-bad fall back value, as it allows processing to continue.  If you want to make sure you don't have intermixed good and bad results, NaN will at least block the bad results from contaminating things.  
I also wanted to raise the point that this should become obsolete soon, as the transition to the new ISR task will switch all `postISRCCD` images to electrons.  I then scrolled down and saw that this method is already getting units, and should therefore be future-proof (and the choice of NaN or 1.0 ignored because it will always use the 1.0 case from line 714 below).

Reply:
Thanks @czwa! Just to confirm, it sounds like you are ok with returning a NaN if the gain is not set? I do not expect this to happen in any cases where we are processing real data, and really this has only been necessary to deal with test cases.  (And yes, understanding the current/future units issue was one of the things that prompted me to dig into the units of the readNoise in ISR.)